src/main/scala/org/constellation/storage/SnapshotService.scala

```diff
@@ -95,27 +96,38 @@ class SnapshotService[F[_]: Concurrent](
               nextSnapshot.lastSnapshot,
               nextHeightInterval - snapshotHeightInterval,
               predictedReputation
-            )
+          )
         )
       )
     } yield created
 
-  def writeSnapshotInfoToDisk: EitherT[F, SnapshotInfoIOError, Unit] =
+  def writeSnapshotPart(path: String, part: Array[Byte]): F[Unit] =
+    Resource
+      .fromAutoCloseable(Sync[F].delay(new FileOutputStream(path)))
+      .use(
+        stream =>
+          Sync[F].delay {
+            stream.write(part)
+          }.flatTap { _ =>
+            logger.debug(s"SnapshotInfo part written for path: $path")
+        }
+      )
+
+  def snapshotInfoWriterProc(plan: Seq[(String, Array[Byte])], basePath: String) = plan.toList.parTraverse {
+    case (path, part) => writeSnapshotPart(File(basePath, path).pathAsString, part)
+  }
+
+  def writeSnapshotInfoParts(info: SnapshotInfo, basePath: String = dao.snapshotInfoPath.pathAsString) =
+    info.toSnapshotInfoSer().write[F[List[Unit]]](snapshotInfoWriterProc)(basePath = basePath)
+
+  def writeSnapshotInfoToDisk(overWritePath: String = ""): EitherT[F, SnapshotInfoIOError, Unit] =
     EitherT.liftF {
       getSnapshotInfoWithFullData.flatMap { info =>
         if (info.snapshot == Snapshot.snapshotZero) Sync[F].unit
         else {
-          val path = dao.snapshotInfoPath.pathAsString
-          Resource
-            .fromAutoCloseable(Sync[F].delay(new FileOutputStream(path)))
-            .use(
-              stream =>
-                Sync[F].delay {
-                  stream.write(KryoSerializer.serializeAnyRef(info))
-                }.flatTap { _ =>
-                  logger.debug(s"SnapshotInfo written for hash: ${info.snapshot.hash} in path: ${path}")
-                }
-            )
+          for {
+            _ <- writeSnapshotInfoParts(info, overWritePath)
```

Comment:
We can write more simply `writeSnapshotInfoParts(info, overWritePath).map(_ => ())`